### PR TITLE
[lua] [sql] Audit & fix Altepa Gate

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Altepa Gate was broken and didn't reset properly. This PR fixes it and adds some QoL for players to see the state of the columns better.

## Steps to test these changes
Use Altepa Gate, see it reset properly, have other players see the changes from a long distance.

### CODE REVIEW

**Review by Xaver-DaRed — APPROVED** [review]
LGTM

Out of curiosity, do any other places may require a similar method, off the top of your head?

**Comment by WinterSolstice8** [comment]
> Out of curiosity, do any other places may require a similar method, of the top of your head?

Altepa gate is wholly unique afaik? not sure, not many switches that you can see from a distance to begin with.

**Comment by Xaver-DaRed** [comment]
Oh well. better then.

Was gonna say that if this kind of logic could be used in other places, it may justify an npc utility, or something similar.

**Review by TeoTwawki — APPROVED** [review]

File: scripts/zones/Western_Altepa_Desert/npcs/_3h5.lua [hunk 1]
```diff
@@ -25,9 +28,19 @@ entity.onTrigger = function(player, npc)
...
         GetNPCByID(ID.npc.ALTEPA_GATE + 8):getAnimation() == xi.anim.OPEN_DOOR
     then
-        local openTime = math.random(15, 30) * 60
+        local openTime = math.random(15, 30) * 1000 * 60
+
         for i = ID.npc.ALTEPA_GATE, ID.npc.ALTEPA_GATE + 8 do
-            GetNPCByID(i):openDoor(openTime)
+            local door = GetNPCByID(i)
+
+            -- openDoor used to (read: a decade ish ago before Sol) allow calls to doors already open and set a timer to close them
```
Comment (TeoTwawki) [inline scripts/zones/Western_Altepa_Desert/npcs/_3h5.lua:36]: side note: want that functionality back :( 

likewise closeDoor used to work on already closed doors and then open them after the time ended.

**Comment by cocosolos** [comment]
> LGTM
> 
> Out of curiosity, do any other places may require a similar method, off the top of your head?

Port Bastok bridge uses something similar I think.

**Comment by WinterSolstice8** [comment]
> Port Bastok bridge uses something similar I think.

Yeah, I had updated that a while ago to send an update on how the bridge looks to the entire zone. It's not being done on zone in -- so it could still be improved.

Honestly, I think things like this should be inserted into the players SpawnNPCLists permanently so we don't need code-specific updates. There's a bunch of weird stuff if you're not in range to get updates. Depending on where you are, you sometimes don't see the airship's propellors for instance. I haven't tried tracking that one down because it's purely cosmetic and not functional (and doesn't alter _human_ behavior like the bridge or altepa gate would)

**Comment by cocosolos** [comment]
> Honestly, I think things like this should be inserted into the players SpawnNPCLists permanently so we don't need code-specific updates.

This is probably the way to go. I looked at Altepa gate a few months ago and noticed the same thing you did where if you get close then far again, the column reverts. This is good enough for now though. 👍

**Pull request merged**